### PR TITLE
[WIP] storage: change token fetch to POST request

### PR DESCRIPTION
**Pull request by imeoer**
Try to make compatible with more registries, for example Google Artifact Registry.

## Details
Try to make compatible with more registries, for example Google Artifact Registry.

## Types of changes

_What types of changes does your PullRequest introduce? Put an `x` in all the boxes that apply:_

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

_Go over all the following points, and put an `x` in all the boxes that apply._

- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: storage/src/backend/registry.rs [hunk 1]
```diff
@@ -260,7 +260,7 @@ impl RegistryState {
 
         let token_resp = connection
             .call::<&[u8]>(
-                Method::GET,
+                Method::POST,
```
Comment (changweige) [inline storage/src/backend/registry.rs:263]: Could you please provide some references proving POST method is more generic?  They are too different. 🤣

Comment (imeoer) [reply storage/src/backend/registry.rs:263]: Will ref the [codes](https://github.com/containerd/containerd/blob/878132923db35f31c8c7b324d9cd1e5f59046064/remotes/docker/auth/fetch.go#L117), but we still need more [tests](https://github.com/dragonflyoss/image-service/pull/764/files).

**Comment by lilic** [comment]
Hello 👋 Great project, really interested in using this in Google Artifact Registry! I am just curious if there is a reason this is still in WIP PR and if it will be picked up? Thank you!

File: storage/src/backend/registry.rs (hunk 1 above)
Comment (philandstuff) [reply storage/src/backend/registry.rs:263]: from https://docs.docker.com/registry/spec/auth/oauth/

> Not all token servers implement oauth2. If the request to the endpoint returns 404 using the HTTP POST method, refer to [Token Documentation](https://docs.docker.com/registry/spec/auth/token/) for using the HTTP GET method supported by all token servers.

**Comment by imeoer** [comment]
Fixed in master.

**Comment by imeoer** [comment]
@lilic Should be fixed in https://github.com/dragonflyoss/image-service/pull/1425

**Pull request closed without merging**